File: Data Structures & Algorithms/pow-x-n/submission-0.java
class Solution {
    public double myPow(double x, int n) {
        if(x == 0 || n == 1) return x;
        if(n == 0) return 1;
        
        double pow = myPow(x, Math.abs(n/2));
        double res;

        if(Math.abs(n % 2) == 1) res = pow*pow*x;
        else res = pow*pow;

        System.out.println(res);

        if(n > 0) return res;
        else    {
            return 1/res;
        }
    }
}

File: Data Structures & Algorithms/eating-bananas/submission-0.py
class Solution:
    def minEatingSpeed(self, piles: List[int], h: int) -> int:
        res = sum(piles)
        l, r = 1, max(piles)
        

        while l <= r:
            mid = (r + l) // 2
            time = self.timeToEat(piles, mid)

            if time > h:
                l = mid + 1
            else:
                r = mid - 1
                res = min(res, mid)

        return res
    

    def timeToEat(self, piles: List[int], k: int) -> int:
        time = 0
        for p in piles:
            time += math.ceil(p / k)
        
        return time

